plugin/cassandra/plugin.go

```diff
@@ -0,0 +1,666 @@
+			},
+			plugin.Field{
+				Mode:     "target",
+				Name:     "cassandra_datadir",
+				Type:     "abspath",
+				Title:    "Path to Cassandra data/ directory",
+				Help:     "The absolute path to the data/ directory that contains the Cassandra database files.",
+				Default:  "/var/vcap/store/cassandra/data",
+				Required: true,
+			},
+			plugin.Field{
+				Mode:     "target",
+				Name:     "cassandra_tar",
+				Type:     "abspath",
+				Title:    "Path to `tar` utility",
+				Help:     "By default, the plugin will search the local `$PATH` to find the `tar` utility.",
+				Default:  DefaultTar,
+				Required: true,
+			},
+		},
+	}
+
+	plugin.Run(p)
+}
+
+type CassandraPlugin plugin.PluginInfo
+
+type CassandraInfo struct {
+	Host     string
+	Port     string
+	User     string
+	Password string
+	Keyspace string
+	BinDir   string
+	DataDir  string
+	Tar      string
+}
+
+// This function should be used to return the plugin's PluginInfo, however you decide to implement it
+func (p CassandraPlugin) Meta() plugin.PluginInfo {
+	return plugin.PluginInfo(p)
+}
+
+// Called to validate endpoints from the command line
+func (p CassandraPlugin) Validate(endpoint plugin.ShieldEndpoint) error {
+	var (
+		s    string
+		err  error
+		fail bool
+	)
+
+	s, err = endpoint.StringValueDefault("cassandra_host", "")
+	if err != nil {
+		fmt.Printf("@R{\u2717 cassandra_host          %s}\n", err)
+		fail = true
+	} else if s == "" {
+		fmt.Printf("@G{\u2713 cassandra_host}          using default node @C{%s}\n", DefaultHost)
+	} else {
+		fmt.Printf("@G{\u2713 cassandra_host}          @C{%s}\n", s)
+	}
+
+	s, err = endpoint.StringValueDefault("cassandra_port", "")
+	if err != nil {
+		fmt.Printf("@R{\u2717 cassandra_port          %s}\n", err)
+		fail = true
+	} else if s == "" {
+		fmt.Printf("@G{\u2713 cassandra_port}          using default port @C{%s}\n", DefaultPort)
+	} else {
+		fmt.Printf("@G{\u2713 cassandra_port}          @C{%s}\n", s)
+	}
+
+	s, err = endpoint.StringValueDefault("cassandra_user", "")
+	if err != nil {
+		fmt.Printf("@R{\u2717 cassandra_user          %s}\n", err)
+		fail = true
+	} else if s == "" {
+		fmt.Printf("@G{\u2713 cassandra_user}          using default user @C{%s}\n", DefaultUser)
+	} else {
+		fmt.Printf("@G{\u2713 cassandra_user}          @C{%s}\n", s)
+	}
+
+	s, err = endpoint.StringValueDefault("cassandra_password", "")
+	if err != nil {
+		fmt.Printf("@R{\u2717 cassandra_password      %s}\n", err)
+		fail = true
+	} else if s == "" {
+		fmt.Printf("@G{\u2713 cassandra_password}      using default password @C{%s}\n", DefaultPassword)
+	} else {
+		fmt.Printf("@G{\u2713 cassandra_password}      @C{%s}\n", s)
+	}
+
+	s, err = endpoint.StringValue("cassandra_keyspace")
+	if err != nil {
+		fmt.Printf("@R{\u2717 cassandra_keyspace      %s}\n", err)
+		fail = true
+	} else {
+		fmt.Printf("@G{\u2713 cassandra_keyspace}      @C{%s}\n", s)
+	}
+
+	s, err = endpoint.StringValueDefault("cassandra_bindir", "")
```

Comment:
In SHIELD v8 and beyond, we now set the `$PATH` environment variable more explicitly, to free up the plugin authors from having to code in explicit path support for executing binaries, and to free up SHIELD users from having to micromanage where they installed all the things.  That responsibility now lies with the team(s) deploying SHIELD agent to the places.

We should therefore remove the `cassandra_bindir` property and just rely on `$PATH`.

See https://github.com/starkandwayne/shield-boshrelease/blob/master/jobs/shield-agent/spec#L49-L52

Reply:
Makes sense indeed. So we'll provide the `/var/vcap/cassandra/job/bin` directory to the SHIELD agent through this `env.path` list.